user-creation/src/main/java/no/unit/nva/useraccessservice/usercreation/UserEntriesCreatorForPerson.java

```diff
@@ -104,8 +104,7 @@ private UserDto updateUserAffiliation(UserDto existingUser,
                                           Person person) throws NotFoundException {
         var affiliation = person.getConsistentUnitAffiliation(existingUser.getInstitutionCristinId());
         var updatedUser = existingUser.copy().withAffiliation(affiliation).build();
-        identityService.updateUser(updatedUser);
-        return updatedUser;
+        return identityService.updateUser(updatedUser);
```

Comment:
Nor returning the updated user so that we detect new access rights during login.